components/tikv_alloc/src/lib.rs

```diff
@@ -91,223 +92,23 @@
-// Rust. Afterwards this might be removed.
-#[cfg(not(all(unix, not(fuzzing), not(feature = "no-jemalloc"))))]
-#[global_allocator]
-static ALLOC: std::alloc::System = std::alloc::System;
-
-pub use self::imp::*;
+#[cfg(not(all(unix, not(fuzzing), feature = "jemalloc")))]
+mod default;
 
 pub type AllocStats = Vec<(&'static str, usize)>;
 
-// The implementation of this crate when jemalloc is turned on
-#[cfg(all(unix, not(fuzzing), not(feature = "no-jemalloc")))]
-mod imp {
-    use super::AllocStats;
-    use jemalloc_ctl::{stats, Epoch as JeEpoch};
-    use jemallocator::ffi::malloc_stats_print;
-    use libc::{self, c_char, c_void};
-    use std::{io, ptr, slice};
-
-    pub use self::profiling::dump_prof;
-
-    pub fn dump_stats() -> String {
-        let mut buf = Vec::with_capacity(1024);
-        unsafe {
-            malloc_stats_print(
-                write_cb,
-                &mut buf as *mut Vec<u8> as *mut c_void,
-                ptr::null(),
-            )
-        }
-        String::from_utf8_lossy(&buf).into_owned()
-    }
-
-    pub fn fetch_stats() -> io::Result<Option<AllocStats>> {
-        // Stats are cached. Need to advance epoch to refresh.
-        JeEpoch::new()?.advance()?;
-
-        Ok(Some(vec![
-            ("allocated", stats::allocated()?),
-            ("active", stats::active()?),
-            ("metadata", stats::metadata()?),
-            ("resident", stats::resident()?),
-            ("mapped", stats::mapped()?),
-            ("retained", stats::retained()?),
-        ]))
-    }
-
-    #[allow(clippy::cast_ptr_alignment)]
-    extern "C" fn write_cb(printer: *mut c_void, msg: *const c_char) {
-        unsafe {
-            // This cast from *c_void to *Vec<u8> looks like a bad
-            // cast to clippy due to pointer alignment, but we know
-            // what type the pointer is.
-            let buf = &mut *(printer as *mut Vec<u8>);
-            let len = libc::strlen(msg);
-            let bytes = slice::from_raw_parts(msg as *const u8, len);
-            buf.extend_from_slice(bytes);
-        }
-    }
-
-    #[cfg(test)]
-    mod tests {
-        #[test]
-        fn dump_stats() {
-            assert_ne!(super::dump_stats().len(), 0);
-        }
-    }
-
-    #[cfg(feature = "mem-profiling")]
-    mod profiling {
-        use std::ffi::CString;
-        use std::{env, ptr};
-
-        use jemallocator;
-        use libc::c_char;
-
-        // C string should end with a '\0'.
-        const PROF_ACTIVE: &'static [u8] = b"prof.active\0";
-        const PROF_DUMP: &'static [u8] = b"prof.dump\0";
-
-        struct DumpPathGuard(Option<Vec<u8>>);
-
-        impl DumpPathGuard {
-            fn from_cstring(s: Option<CString>) -> DumpPathGuard {
-                DumpPathGuard(s.map(|s| s.into_bytes_with_nul()))
-            }
+// Allocators
+#[cfg(all(unix, not(fuzzing), feature = "jemalloc"))]
+#[path = "jemalloc.rs"]
+mod imp;
+#[cfg(all(unix, not(fuzzing), feature = "tcmalloc"))]
+#[path = "tc.rs"]
```

Comment:
Can we consistently name it to `tcmalloc.rs`?

Reply:
@Hoverbear  The name is the same as `tcmalloc` crate.

Reply:
@ZhangHanDong I think Hoverbear is saying that the name of the source file for the tcmalloc implementation, `tc.rs`, is inconsistent with the name of the source file for jemalloc, `jemalloc.rs`. Please do change it.